src/Symfony/Component/MimePgp/PgpEncrypter.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\MimePgp;
+
+use Symfony\Component\Mime\Header\Headers;
+use Symfony\Component\Mime\Header\MailboxListHeader;
+use Symfony\Component\Mime\Message;
+use Symfony\Component\MimePgp\Exception\KeyNotFoundException;
+use Symfony\Component\MimePgp\Mime\Part\Multipart\PgpEncryptedPart;
+use Symfony\Component\MimePgp\Mime\Part\PgpEncryptedInitializationPart;
+use Symfony\Component\MimePgp\Mime\Part\PgpEncryptedMessagePart;
+
+/*
+ * @author PuLLi <the@pulli.dev>
+ */
+final readonly class PgpEncrypter extends PgpMime
+{
+    /**
+     * @param array<string, string>                                                                              $recipientKeys
+     * @param array{binary?: string, cipher_algorithm?: string, digest_algorithm?: string, temp_prefix?: string} $options
+     */
+    public function __construct(private array $recipientKeys, array $options = [])
```

Comment:
instead of a static map of recipient keys, I think we need an interface `getRecipientKey(string $email)`, to allow different implementations. I'm quite sure many projects will need to load the recipient key from a dynamic source (the database for instance) instead of being able to configure the keys for all their recipients in the DI configuration.

Reply:
The design is very similar to `SMimeEncrypter` for which you can pass a list of certificate paths.
This is a static map because when you send an encrypted email, you have to select the recipient public key at first.
Then you can create the `PGPEncrypter` object by passing those keys.

Reply:
Creating the PGPEncrypter specifically for each email being encrypted won't be compatible with the encrypter event listener registered in the mailer, as that involves creating an encrypter that will be used for any sent email.